### PR TITLE
Add Technical Indicator Endpoints

### LINKED ISSUE #290: Client update needed to match REST spec changes
A diff between this client library's spec and our hosted spec was found. The client may need an update to match any changes in our API. See the diff below:
```diff
--- https://raw.githubusercontent.com/polygon-io/client-python/master/.polygon/rest.json
+++ https://api.polygon.io/openapi
@@ -14981,7 +14981,15 @@
                 "LT",
                 "ST"
               ],
-              "type": "string"
+              "type": "string",
+              "x-polygon-go-field-tags": {
+                "tags": [
+                  {
+                    "key": "binding",
+                    "value": "required"
+                  }
+                ]
+              }
             }
           },
           {
@@ -15281,15 +15289,39 @@
                         "properties": {
                           "cash_amount": {
                             "description": "The cash amount of the dividend per share owned.",
-                            "type": "number"
+                            "type": "number",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "currency": {
                             "description": "The currency in which the dividend is paid.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "declaration_date": {
                             "description": "The date that the dividend was announced.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "dividend_type": {
                             "description": "The type of dividend. Dividends that have been paid and/or are expected to be paid on consistent schedules are denoted as CD.\nSpecial Cash dividends that have been paid that are infrequent or unusual, and/or can not be expected to occur in the future are denoted as SC.\nLong-Term and Short-Term capital gain distributions are denoted as LT and ST, respectively.",
@@ -15299,27 +15325,75 @@
                               "LT",
                               "ST"
                             ],
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "ex_dividend_date": {
                             "description": "The date that the stock first trades without the dividend, determined by the exchange.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "frequency": {
                             "description": "The number of times per year the dividend is paid out.  Possible values are 0 (one-time), 1 (annually), 2 (bi-annually), 4 (quarterly), and 12 (monthly).",
-                            "type": "integer"
+                            "type": "integer",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "pay_date": {
                             "description": "The date that the dividend is paid out.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "record_date": {
                             "description": "The date that the stock must be held to receive the dividend, set by the company.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           },
                           "ticker": {
                             "description": "The ticker symbol of the dividend.",
-                            "type": "string"
+                            "type": "string",
+                            "x-polygon-go-field-tags": {
+                              "tags": [
+                                {
+                                  "key": "binding",
+                                  "value": "required"
+                                }
+                              ]
+                            }
                           }
                         },
                         "required": [
@@ -15401,15 +15460,39 @@
                 "properties": {
                   "cash_amount": {
                     "description": "The cash amount of the dividend per share owned.",
-                    "type": "number"
+                    "type": "number",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "currency": {
                     "description": "The currency in which the dividend is paid.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "declaration_date": {
                     "description": "The date that the dividend was announced.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "dividend_type": {
                     "description": "The type of dividend. Dividends that have been paid and/or are expected to be paid on consistent schedules are denoted as CD.\nSpecial Cash dividends that have been paid that are infrequent or unusual, and/or can not be expected to occur in the future are denoted as SC.\nLong-Term and Short-Term capital gain distributions are denoted as LT and ST, respectively.",
@@ -15419,27 +15496,75 @@
                       "LT",
                       "ST"
                     ],
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "ex_dividend_date": {
                     "description": "The date that the stock first trades without the dividend, determined by the exchange.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "frequency": {
                     "description": "The number of times per year the dividend is paid out.  Possible values are 0 (one-time), 1 (annually), 2 (bi-annually), 4 (quarterly), and 12 (monthly).",
-                    "type": "integer"
+                    "type": "integer",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "pay_date": {
                     "description": "The date that the dividend is paid out.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "record_date": {
                     "description": "The date that the stock must be held to receive the dividend, set by the company.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   },
                   "ticker": {
                     "description": "The ticker symbol of the dividend.",
-                    "type": "string"
+                    "type": "string",
+                    "x-polygon-go-field-tags": {
+                      "tags": [
+                        {
+                          "key": "binding",
+                          "value": "required"
+                        }
+                      ]
+                    }
                   }
                 },
                 "required": [
@@ -15477,15 +15587,39 @@
                       "properties": {
                         "cash_amount": {
                           "description": "The cash amount of the dividend per share owned.",
-                          "type": "number"
+                          "type": "number",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "currency": {
                           "description": "The currency in which the dividend is paid.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "declaration_date": {
                           "description": "The date that the dividend was announced.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "dividend_type": {
                           "description": "The type of dividend. Dividends that have been paid and/or are expected to be paid on consistent schedules are denoted as CD.\nSpecial Cash dividends that have been paid that are infrequent or unusual, and/or can not be expected to occur in the future are denoted as SC.\nLong-Term and Short-Term capital gain distributions are denoted as LT and ST, respectively.",
@@ -15495,27 +15623,75 @@
                             "LT",
                             "ST"
                           ],
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "ex_dividend_date": {
                           "description": "The date that the stock first trades without the dividend, determined by the exchange.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "frequency": {
                           "description": "The number of times per year the dividend is paid out.  Possible values are 0 (one-time), 1 (annually), 2 (bi-annually), 4 (quarterly), and 12 (monthly).",
-                          "type": "integer"
+                          "type": "integer",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "pay_date": {
                           "description": "The date that the dividend is paid out.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "record_date": {
                           "description": "The date that the stock must be held to receive the dividend, set by the company.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         },
                         "ticker": {
                           "description": "The ticker symbol of the dividend.",
-                          "type": "string"
+                          "type": "string",
+                          "x-polygon-go-field-tags": {
+                            "tags": [
+                              {
+                                "key": "binding",
+                                "value": "required"
+                              }
+                            ]
+                          }
                         }
                       },
                       "required": [

```

### PR DESCRIPTION
**Pull request by aitzkovitz**
- initial models
- wip
- restore websocket files
- restore examples to master
- touch up from_dict methods, add tests
- individual result types
- lint

closes #290

### CODE REVIEW

File: polygon/rest/indicators.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        window: Optional[int] = None,
+        adjusted: Optional[bool] = None,
+        expand_underlying: Optional[bool] = None,
+        order: Optional[Union[str, Order]] = None,
+        params: Optional[Dict[str, Any]] = None,
+        series_type: Optional[Union[str, SeriesType]] = None,
+        raw: bool = False,
+    ) -> Union[RSIIndicatorResults, HTTPResponse]:
+        """
+        Get SMA values for a given ticker over a given range with the specified parameters
```
Comment (aitzkovitz) [inline polygon/rest/indicators.py:141]: nit: this should be EMA

File: polygon/rest/indicators.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        signal_window: Optional[int] = None,
+        adjusted: Optional[bool] = None,
+        expand_underlying: Optional[bool] = None,
+        order: Optional[Union[str, Order]] = None,
+        params: Optional[Dict[str, Any]] = None,
+        series_type: Optional[Union[str, SeriesType]] = None,
+        raw: bool = False,
+    ) -> Union[MACDIndicatorResults, HTTPResponse]:
+        """
+        Get SMA values for a given ticker over a given range with the specified parameters
```
Comment (aitzkovitz) [inline polygon/rest/indicators.py:196]: this should be macd, not sma

**Comment by aitzkovitz** [comment]
@danielatpolygonio are we good to merge this in or mark it as ready for review? If not i can make the remaining changes

**Comment by danielatpolygonio** [comment]
I think this is ready to go.

**Comment by aitzkovitz** [comment]
whoops looks like we still need to remove multipler, i'll do that

File: polygon/rest/models/__init__.py [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
-from .contracts import *
+from .indicators import (
+    SMAIndicatorResults,
+    EMAIndicatorResults,
+    RSIIndicatorResults,
+    MACDIndicatorResults,
+    IndicatorValue,
+    MACDIndicatorValue,
+    Underlying as IndicatorUnderlying,
+)
```
Comment (clickingbuttons) [inline polygon/rest/models/__init__.py:16]: Why not `import *`?

**Review by clickingbuttons — COMMENTED** [review]
Other than this looks good. Thanks for the work!

File: polygon/rest/indicators.py (hunk 2 above)
Comment (clickingbuttons) [reply polygon/rest/indicators.py:189]: +1

File: polygon/rest/models/__init__.py (hunk 3 above)
Comment (danielatpolygonio) [reply polygon/rest/models/__init__.py:16]: We have to rename Underlying to IndicatorUnderlying. I personally don't love import * in general

Comment (clickingbuttons) [reply polygon/rest/models/__init__.py:16]: Oh, why not just rename `Underlying` to `IndicatorUnderlying` then?

**Review by clickingbuttons — APPROVED** [review]
Ty!

**Review by morningvera — CHANGES_REQUESTED** [review]
we still need to update the rest spec for this: `make rest-spec`

**Review by morningvera — APPROVED** [review]
lgtm!

**Pull request merged**